source/reflectionzeug/source/JsonPropertySerializer.cpp

```diff
@@ -0,0 +1,101 @@
+#include <reflectionzeug/JsonPropertySerializer.h>
+
+#include <iostream>
+
+#include <loggingzeug/logging.h>
+
+#include <reflectionzeug/AbstractValueProperty.h>
+#include <reflectionzeug/JsonSerializationVisitor.h>
+#include <reflectionzeug/ValueProperty.h>
+#include <reflectionzeug/PropertyGroup.h>
+#include <reflectionzeug/util.h>
+
+using namespace loggingzeug;
+
+namespace reflectionzeug
+{
+    
+JsonPropertySerializer::JsonPropertySerializer()
+    : m_nestingLevel(0)
+{
+    m_serialize =
+        [this](PropertyGroup & group)
+        {
+            m_elementCount[m_nestingLevel]--;
+
+            m_fstream << indent(m_nestingLevel);
+            if (!group.name().empty())
+            {
+                m_fstream << "\"" << group.name() << "\": ";
+            }
+            m_fstream << "{" << std::endl;
+
+            m_nestingLevel++;
+            m_elementCount.push_back(group.asCollection()->count());
+
+            group.forEachValue(
+                [this](AbstractValueProperty & property)
+                {
+                    m_elementCount[m_nestingLevel]--;
+                    serializeValue(property);
+                }
+            );
+
+            group.forEachGroup(m_serialize);
+            m_nestingLevel--;
+            m_fstream << indent(m_nestingLevel) << "}";
+            if (m_elementCount.at(m_nestingLevel) > 0)
+            {
+                m_fstream << ",";
+            }
+            m_fstream << std::endl;
+            m_elementCount.pop_back();
+        };
+}

```

Comment:
i do not understand why it is necessary to use lambda expressions here - why not just use functions?
Especially: this is a recursive use of lambdas? again, why not use plain old functions ...